addons/hr_attendance/models/hr_employee.py

```diff
@@ -0,0 +1,61 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import api, fields, models, _
+from odoo.exceptions import UserError
+
+class HrEmployee(models.Model):
+    _inherit = "hr.employee"
+    _description = "Employee"
+
+    state = fields.Selection([('absent', 'Absent'), ('present', 'Present')], compute='_state', string='Attendance')
+    last_sign = fields.Datetime(compute='_last_sign', string='Last Sign')
+    attendance_access = fields.Boolean(compute='_attendance_access', string='Attendance Access')
+
+    @api.multi
+    def _state(self):
+        self.env.cr.execute('SELECT hr_attendance.action, hr_attendance.employee_id \
+                FROM ( \
+                    SELECT MAX(name) AS name, employee_id \
+                    FROM hr_attendance \
+                    WHERE action in (\'sign_in\', \'sign_out\') \
+                    GROUP BY employee_id \
+                ) AS foo \
+                LEFT JOIN hr_attendance \
+                    ON (hr_attendance.employee_id = foo.employee_id \
+                        AND hr_attendance.name = foo.name) \
+                WHERE hr_attendance.employee_id IN %s', (tuple(self.ids),))
+        for res in self.env.cr.fetchall():
+            self.state = 'present' if res[0] == 'sign_in' else 'absent'
+
+    def _last_sign(self):
+        attendance = self.env['hr.attendance'].search(['&', ('employee_id', '=', self.id), ('action', 'in', ('sign_in', 'sign_out'))], order='name')
+        for attend in attendance:
+            self.last_sign = attend.name
```

Comment:
Refactor this method:
do something: Prepare one dict something like {attend.id: attend.name, ......}
and then iterate over self and pick the id from created dict and assign it to last_sign